source/particle/world.cc

```diff
@@ -54,30 +54,7 @@ namespace aspect
     {
       CitationInfo::add("particles");
       if (particle_load_balancing & ParticleLoadBalancing::repartition)
-#if DEAL_II_VERSION_GTE(9,4,0)
-        this->get_triangulation().signals.weight.connect(
-#else
-        this->get_triangulation().signals.cell_weight.connect(
-#endif
```

Comment:
This isn't right. The idea for the change is that ASPECT now requires deal.II 9.4 or later, and so the `#else` branch can't happen any more. But you need to keep the `#if` branch, along with all of the code you removed from below.